doc: samples: can: update sample path for mcp2515 board and isotp sample code block

str4t0m opened the pull request:
Update the path of the can counter sample in the mcp2515 shild docs,
as well as the path of the subsys/canbus/isotp code example in the docs.

Fixes: #51752

henrikbrixandersen reviewed (changes requested):
Looks good, but please split into two commits.

str4t0m committed:
boards: shields: mcp2515: doc: update sample path

The path of the sample has changed in the meantime, update this in
the docs of the board.

Signed-off-by: Thomas Stranger <thomas.stranger@outlook.com>

str4t0m committed:
samples: subsys: canbus: isotp: fix sample path in Readme.rst

The path in the readme did not match the sample's real path.

Signed-off-by: Thomas Stranger <thomas.stranger@outlook.com>

str4t0m commented:
I have now split it into 2 commits.

henrikbrixandersen reviewed (approved):
Thanks!
